### DISCUSSION TITLE
Alpinejs 3.14.4 breaks attribute binding with blade @if() in Livewire 2

### DISCUSSION THREAD

**Discussion by basepack** [alpinejs/alpine#4470] [7. Bugs (archived)]
After updating Alpine.js from version 3.14.3 to 3.14.4 (or higher), using Blade conditionals like @if($this->order) within a Livewire component's blade file are broken if $this->order is null when the component mounted. 

I use Livewire 2.0, but I do not know if that makes a difference. It seems that the newer versions of the AlpineJS dom-diffing are incompatible from versions higher then 3.14.3 and are preventing attributes from being correctly set when using @if conditionals. Removing the @if directive restores the expected behavior.

**Steps to Reproduce**
- Use Alpine.js 3.14.4 with Livewire 2.0.
- Implement a Blade @if($this->order) conditional within an Alpine.js component that sets attributes based on the condition.
- Emit the show-modal event from the Livewire component.
- See that open attribute is not set when the @if directive is present.

Things tried that did not work:
- Adding wire:key in various places
- Removing the wire:click

Of course, the production code is much more complicated, but the below example is a working example of this bug.

**Example code that fails since 3.14.4**
```blade
<div x-data="{ open: false }"
     x-show="open"
     x-init="
        @this.on('show-modal', () => {
            open = true;
        })
    ">
    
    {{-- a surrounding div already added for dom-diffing --}}
    <div class="p-4">
        <!-- Just remove the below if and it works in AlpineJS > 3.14.3 -->
        <!-- Belof if setup did work in AlpineJS 3.14.3 and lower -->
        @if($this->order) 
            <div wire:click="save" x-bind:class="{ 'active': open }">Save</div>
        @endif
    </div>
</div>
```

```php
use Livewire\Component;

class OrderComponent extends Component
{
      public ?Order $order = null;

    protected $listeners = [
        'showModal',
    ];

    public function showModal(int $orderId)
    {
        $this->order = Order::find($orderId);

        $this->emitSelf('show-modal');
    }

    public function save()
    {
        //
    }

    public function render()
    {
        return view('livewire.modal-component');
    }
}
``` 

So the gist of it is:
- The $this->order property is null when the component initializes
- After an event $this->order is set
- The blade @if($this->order) does not work anymore after upgrading to AlpineJS > 3.14.3

**Comment by danie-ramdhani** [comment]
I think it fixed in latest version (3.14.7)

**Reply by basepack** [reply]
I just tested it, but it is not unfortunately.

**Reply by dlgombert** [reply]
Switching to 3.14.7 fixed it for me.  After I had pulled out all of my hair and am now bald.  Slightly different issue, but similar enough.

**Comment by SimoTod** [comment]
`@if` runs on the server, it seems unlikely that it will affect alpine that runs on the browser. Are you sure it's not something else?

I used this script since your initial example was not fully testable:
```php
<?php

namespace App\Livewire;

use DateTime;
use Livewire\Component;

class Test extends Component
{
    public ?DateTime $date = null;

    public function test(): void
    {
        $this->date = new DateTime();
        $this->dispatch('show-modal');
    }

    public function save()
    {
        //
    }

    public function render()
    {
        return view('livewire.test');
    }
}
```
```html
// test.blade.php
<div>
    <button type="button" wire:click="test">Test</button>
    <div x-data="{ open: false }"
         x-show="open"
         x-init="
            @this.on('show-modal', () => {
                open = true;
            })
        ">
        <div class="p-4">
            @if($this->date)
                <div wire:click="save" x-bind:class="{ 'active': open }">Save</div>
            @endif
        </div>
    </div>
</div>
```
after clicking test, I can see `Save` and the active class is correctly bound.

**Reply by Yinci** [reply]
It seems that Livewire rendered content is not triggering AlpineJS updates (V3.14.4 through 3.14.7). We have a component that renders a list from Livewire, and all the handlers (such as click, class binding) are simply not bound, so it doesn't do anything. Currently trying to find out how to force Alpine to re-check the component...

**Reply by basepack** [reply]
Yea, for us it does still not work with the newer versions and having similar problems. While it worked before the update.

**Reply by Yinci** [reply]
@basepack I wasn't able to find a workaround so for now we've fixed our version to 3.14.3 hoping a newer version fixes this. Not looking forward to having to redo all my Livewire/Alpine components because it can't re-bind event handlers. Pretty annoying this change is in a patch version.

**Reply by basepack** [reply]
For us it is the same, having many modals that are also affected by this issue. We've temporarily reverted to 3.14.3 as well and hoping they fix this in an upcoming version.

**Reply by karan218** [reply]
Have you got any solutions on above issue? Like pop and @if conditions

**Reply by karan218** [reply]
I had same problem with alpine 3.14.9 and livewire v2

**Reply by noefi-com** [reply]
i have the same problem with phoenix liveview, when adding alpinejs components to the dom

**Reply by ciaw** [reply]
Ended up here as I tried updating AlpineJS from 3.14.1 to 3.14.9 and my x-show command broke on div inside a blade @if command just like OP described.

I then tried upgrading to each Alpine JS version after 3.14.1 and it started with 3.14.4. Downgraded to 3.14.3 and everything works again.

**Reply by wannesds** [reply]
Still having same issue. It seems like its init related, once you toggle between the state then x-show starts working after the second toggle.
Multiple lines in using :class breaks aswell, editing the class inline in the browser then suddenly _snaps_ it into its desired state.

@ciaw 
I'v found a workaround for the x-show bug. add a `hidden` class on the div and a `:class="{ '!flex': true}"`  with true as example command/variable
